### PR TITLE
Arb

### PR DESCRIPTION
**Pull request by mseri**
New formula for Arb 2.5.0 and head. 

Arb is a fast C library for arbitrary-precision numerics with rigorous error bounds.
See http://fredrikj.net/arb/ or https://github.com/fredrik-johansson/arb

### CODE REVIEW

File: arb.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  # Will be enabled once the new stable (with patched tests) is released
+  # some of the tests in 2.5.0 are broken because they call not yet implemented
+  # methods
+  #option "with-check", "Enable build-time checking (not recommended)"
+
+  def install
+    system "./configure", "--prefix=#{prefix}"
+    # We need to remove this line to have 2.5.0 compiled on OSX
+    # it is fixed in the new version, this line will disappear then
+    system "sed -i.brew.orig 's/$(QUIET_AR) $(AR) rcs libarb.a $(OBJS);//' Makefile"
```
Comment (iml) [inline arb.rb]: Use the `inreplace` method instead of sed.
```
inreplace "Makefile", "$(QUIET_AR) $(AR) rcs libarb.a $(OBJS);", ""

File: arb.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  url "https://github.com/fredrik-johansson/arb/archive/2.5.0.tar.gz"
+  sha1 "ba51573e0c50250bbb16bfca027dc26531347f12"
+  head "https://github.com/fredrik-johansson/arb.git", :branch => "trunk"
+
+  bottle do
+    root_url "https://downloads.sf.net/project/machomebrew/Bottles/science"
+    sha1 "617731bca8e304ff9351c0fbee22ac078bbeee5e" => :yosemite
+    sha1 "ec020b7499a3b7d19fa4a2c44d2abc30d34c21aa" => :mavericks
+    sha1 "29dbbfc5cf472a0922376ec47e40975b9caec6bb" => :mountain_lion
+  end
```
Comment (iml) [inline arb.rb]: Remove the bottle block, not sure where it came from.

File: arb.rb [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+# TODOs: 
+# - Add Arb-FLINT integration in flint formula, setting Arb optional
+# - (?) Add possibility to choose MPIR in place of GMP
+
+class Arb < Formula
+  homepage "http://fredrikj.net/arb/index.html"
+  #tag "math"
+  url "https://github.com/fredrik-johansson/arb/archive/2.5.0.tar.gz"
+  sha1 "ba51573e0c50250bbb16bfca027dc26531347f12"
+  head "https://github.com/fredrik-johansson/arb.git", :branch => "trunk"
```
Comment (iml) [inline arb.rb]: I don't think `trunk` is a branch name.

**Comment by iml** [comment]
```
==> brew style arb
== /usr/local/Library/Taps/Homebrew/homebrew-science/arb.rb ==
C:  9:  3: Missing space after #.
C: 28:  3: Missing space after #.
C: 38:  5: Missing space after #.

1 file inspected, 3 offenses detected
Error: 3 problems in 1 formulae

==> audit problems
arb:
 * `require "formula"` is now unnecessary
 * 3: Trailing whitespace was found

File: arb.rb [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        arb_init(x);
+        arb_const_pi(x, 50 * 3.33);
+        arb_printn(x, 50, 0); printf("\n");
+        printf("Computed with arb-%s\n", arb_version);
+        arb_clear(x);
+
+        return EXIT_SUCCESS;
+      }
+    EOS
+    system ENV.cc, "test.c", "-larb", "-lflint", "-I/usr/local/include/flint", "-o", "test"
```
Comment (iml) [inline arb.rb]: Don't use absolute paths. Something like `"-I#{Formula["flint"].opt_include}/flint"` for external formula. Or `"-I#{include}"` is probably necessary for -larb.

File: arb.rb (hunk 2 above)
Comment (mseri) [inline arb.rb]: I had used `flint` formula as starting point, I think it was there and I forgot to remove. I am fixing the formula following your comments. Thank you very much

**Comment by dpo** [comment]
Also the `require 'formula'` bit is no longer necessary.

**Pull request closed without merging**